### PR TITLE
✨ update AWS MP listing automation

### PR DESCRIPTION
**Pull request by sayaliM0412**
1. feat: `.github/workflows/dependabot-pr-merge-docker-changes.yml`: Auto Merge Dependabot PRs for Dockerfile Changes
2. feat: `.github/workflows/deploy-extension-to-marketplace.yml`: add `workflow_dispatch` and add `dry-run` capability to deploy extension to Marketplace to test
3. fix: `README.md`: update readme

### CODE REVIEW

File: .github/workflows/dependabot-pr-merge-docker-changes.yml
Comment (StevenMassaro) [inline .github/workflows/dependabot-pr-merge-docker-changes.yml:1]: I noticed this as well: https://docs.github.com/en/code-security/dependabot/working-with-dependabot/automating-dependabot-with-github-actions#enable-auto-merge-on-a-pull-request

not sure if it's any easier than what you have here though.

File: .github/workflows/deploy-extension-to-marketplace.yml [hunk 1]
```diff
@@ -40,17 +40,66 @@ jobs:
...
+
+      - name: Submit New Version Request for AWS MP Listing
+        env:
+          ECR_REPOSITORY: ${{ secrets.ECR_REPOSITORY }}
+          IMAGE_TAG: ${{ github.event.inputs.image_tag }}
+          PRODUCT_ID: ${{ secrets.PRODUCT_ID }}
+          AWS_MP_REGISTRY: ${{ secrets.AWS_MP_REGISTRY }}
+        run: |
+          chmod +x ./.github/utils/submit-aws-mp-listing.sh
+          ./.github/utils/submit-aws-mp-listing.sh
```
Comment (StevenMassaro) [inline .github/workflows/deploy-extension-to-marketplace.yml:105]: Do we actually want to submit a new version request during dry run?

**Review by StevenMassaro — APPROVED** [review]

**Review by filipelautert — APPROVED** [review]

File: .github/workflows/deploy-extension-to-marketplace.yml (hunk 1 above)
Comment (sayaliM0412) [reply .github/workflows/deploy-extension-to-marketplace.yml:105]: yes, we do but under a dry run and I need to edit it to "private" listing and not public. That way QA can test it

Comment (sayaliM0412) [reply .github/workflows/deploy-extension-to-marketplace.yml]: more comment about "Private" and "Public" listing : https://datical.atlassian.net/browse/DAT-18744?focusedCommentId=151342

**Pull request merged**